### ISSUE TITLE
No lockfiles rule

### ISSUE THREAD

**Issue by SamVerschueren** [SamVerschueren/clinton#82]
I never use lockfiles for my modules, only for applications to make sure that the application that is built has the exact same dependencies as the one I tested everything with. Maybe we should add this as a warning instead of an error? What if the lockfile is added to `.gitignore`, should we still warn?

// @sindresorhus

**Comment by sindresorhus** [comment]
:+1: Warning and not if it's in .gitignore.

**Comment by sindresorhus** [comment]
The rule should:

- Warn if `package-lock.json` or `yarn.lock` is in the root.
- Warn if there's no `.npmrc` with `package-lock=false`.
- Warn if there's no `yarn.lock` in `.gitignore`.

**Comment by SamVerschueren** [comment]
Good suggestions!

**Comment by silverwind** [comment]
Just adding that yarn can also be [configured](https://github.com/silverwind/dotfiles/blob/master/.yarnrc) to not emit lockfiles on `install`, I hate them with passion 😉.

**Comment by SamVerschueren** [comment]
Because those are just warnings, I guess they shouldn't/can't be auto-fixable?

> Warn if `package-lock.json` or `yarn.lock` is in the root.
Remove lock file

> Warn if there's no `.npmrc` with `package-lock=false`.
Create `.npmrc` with `package-lock=false`

> Warn if there's no `yarn.lock` in `.gitignore`.
Add `yarn.lock` to the root

**Comment by sindresorhus** [comment]
I don't see why you couldn't auto-fix them. What you suggested above sounds fine.

**Comment by SamVerschueren** [comment]
You're right, I was assuming that warnings shouldn't be auto-fixable. But actually they can. Thanks for the feedback!
